src/clusterfuzz/_internal/bot/tasks/utasks/progression_task.py

```diff
@@ -536,16 +554,19 @@ def find_fixed_range(uworker_input):
         max_index -= 1
         continue
       # Only bad build errors are recoverable.
+      error.progression_task_output = uworker_io.ProgressionTaskOutput(
+          last_progression_min=last_progression_min,
+          last_progression_max=last_progression_max,
+          clear_min_max_metadata=clear_min_max_metadata)
       return error
 
     if result.is_crash():
       min_index = middle_index
     else:
       max_index = middle_index
 
-    # TODO(alhijazi): This cannot be done from an untrusted bot.
-    _save_current_fixed_range_indices(testcase_id, revision_list[min_index],
```

Comment:
As far as I can tell, this should only happen when there is a timeout. But you also added it to handle_progression_bad_state_min_max
Why?

Reply:
`_save_current_fixed_range_indices` was saving the changing to the testcase and uploading it directly with `testcase.put()`. We are not doing this anymore, so in order no to lose changes and keep the progress we did so far, we need to save the last_progression_min,last_progression_max values we have in case of error in this loop.